Assets/Source/Components/Actor/ActorBehavior.cs

```diff
@@ -33,23 +31,28 @@ public int BlasterAmmo
             set => _blasterAmmo = value; 
         }
 
-        
+        private TextMeshProUGUI ammoText;
 
         #region Player Stats
-
         public int MaxHealth { get; set; } = 100;
 
         public int MaxBlasterAmmo { get; set; } = 150;
         #endregion
 
-        // todo: delete this mess
-        public override void ComponentUpdate()
+        public override void ComponentAwake()
         {
-            // Temporary way to show the ammo 
-            var tmp = GetRequiredComponent<TextMeshProUGUI>(GetRequiredChild("AmmoPlaceholder", FindOrCreateCanvas()));
-            tmp.SetText(BlasterAmmo.ToString());
-            base.ComponentUpdate();
+            GameObject hud = GetRequiredChild("HUD", FindOrCreateCanvas());
+            GameObject ammoDisplay = GetRequiredChild("AmmoDisplay", hud);
+            GameObject border = GetRequiredChild("Border", ammoDisplay);
+            ammoText = GetRequiredComponent<TextMeshProUGUI>(GetRequiredChild("AmmoPlaceholder", border));
+
+            base.ComponentAwake();
         }
 
+        public void UseAmmoAndSetText()
+        {
+            BlasterAmmo--;
+            ammoText.SetText($"x{BlasterAmmo}");
```

Comment:
Setting ammo should be separate, and the ui should update every frame so that UI updates are decoupled from this method.  If i updating the ammo value via the inspector, it wouldn't get updated right away.

Reply:
ahh yeah that makes more sense. I just wanted an easy way to do this in one action but decoupling makes sense